tracer.go

```diff
@@ -266,3 +264,17 @@ func (t Tags) Apply(o *StartSpanOptions) {
 		o.Tags[k] = v
 	}
 }
+
+// Tag may be passed as a StartSpanOption to add a tag to new spans.
+type Tag struct {
```

Comment:
can we not make Tag also behave like tags in `ext`?  Then instead of

```
ext.SpanKind.Set(span, ext.SpanKindRPCServerEnum)
```

we could do 
```
ext.SpanKindRPCServer.Set(span)
```